components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminService.java

```diff
@@ -316,6 +321,113 @@ public void updateConsumerApplication(OAuthConsumerAppDTO consumerAppDTO) throws
+    }
+
+    /**
+     *
+     * @param consumerKey
+     * @param newState
+     * @throws IdentityOAuthAdminException
+     */
+    public void updateConsumerAppState(String consumerKey, String newState) throws IdentityOAuthAdminException {
+
+        OAuthAppDAO oAuthAppDAO = new OAuthAppDAO();
+        try {
+            if (OAuthServerConfiguration.getInstance().isCacheEnabled()) {
+                OAuthAppDO oAuthAppDO = appInfoCache.getValueFromCache(consumerKey);
+                if(oAuthAppDO != null) {
+                    oAuthAppDO.setState(newState);
+                } else {
+                    oAuthAppDO = oAuthAppDAO.getAppInformation(consumerKey);
+                }
+                appInfoCache.addToCache(consumerKey, oAuthAppDO);
+
+                if (log.isDebugEnabled()) {
+                    log.debug("App state is updated in the cache.");
+                }
+            }
+
+            Properties properties = new Properties();
+            properties.setProperty(OAuthConstants.OAUTH_APP_NEW_STATE, newState);
+            properties.setProperty(OAuthConstants.ACTION_PROPERTY_KEY, OAuthConstants.ACTION_REVOKE);
+            updateAppAndRevokeTokensAndAuthzCodes(consumerKey, properties);
+
+        } catch ( InvalidOAuthClientException | IdentityOAuth2Exception e) {
+            throw new IdentityOAuthAdminException("Error while updating consumer application state", e);
+        }
+    }
+
+    /**
+     *
+     * @param consumerKey
+     * @throws IdentityOAuthAdminException
+     */
+    public void updateOauthSecretKey(String consumerKey) throws IdentityOAuthAdminException {
+
+        OAuthConsumerDAO oAuthConsumerDAO = new OAuthConsumerDAO();
+        String newSecretKey = OAuthUtil.getRandomNumber();
+        if (OAuthServerConfiguration.getInstance().isCacheEnabled()) {
+            CacheEntry clientCredentialDO = new ClientCredentialDO(newSecretKey);
+            OAuthCache.getInstance().addToCache(new OAuthCacheKey(consumerKey), clientCredentialDO);
+            if (log.isDebugEnabled()) {
+                log.debug("Client Secret is updated in the cache.");
+            }
+        }
+
+        Properties properties = new Properties();
+        properties.setProperty(OAuthConstants.OAUTH_APP_NEW_SECRET_KEY, newSecretKey);
+        properties.setProperty(OAuthConstants.ACTION_PROPERTY_KEY, OAuthConstants.ACTION_REGENERATE);
+        updateAppAndRevokeTokensAndAuthzCodes(consumerKey, properties);
+
+    }
+
+    private void updateAppAndRevokeTokensAndAuthzCodes(String consumerKey, Properties properties)  throws IdentityOAuthAdminException {
+        TokenMgtDAO tokenMgtDAO = new TokenMgtDAO();
+
+        try {
+            Set<String> accessTokens = tokenMgtDAO.getActiveTokensForConsumerKey(consumerKey);
+            if (OAuthServerConfiguration.getInstance().isCacheEnabled()) {
+                OAuthCache oauthCache = OAuthCache.getInstance();
+                for(String accessToken : accessTokens) {
```

Comment:
No, from getActiveTokensForConsumerKey(), it always returns an empty HashSet object.